### PR TITLE
IFU rocm5.0 internal testing 2021-11-15

### PR DESCRIPTION
**Pull request by zstreet87**
[conflicts.txt](https://github.com/ROCmSoftwarePlatform/pytorch/files/7549516/conflicts.txt)
[build_log_IFU-rocm5.0-internal-testing_2021-11-16.txt](https://github.com/ROCmSoftwarePlatform/pytorch/files/7549517/build_log_IFU-rocm5.0-internal-testing_2021-11-16.txt)

### CODE REVIEW

**Review by jeffdaily — CHANGES_REQUESTED** [review]
We should use `USE_ROCM` and not `__HIP_PLATFORM_HCC__`.

**Comment by zstreet87** [comment]
@jeffdaily
So you are referring to this change in Macros.h and want to take the upstream change?

++<<<<<<< HEAD
 +#if defined(__ANDROID__) || defined(__APPLE__) || defined(__HIP_PLATFORM_HCC__)
++=======
+ #if defined(__ANDROID__) || defined(__APPLE__) || defined(__XROS__) || \
+     (defined(USE_ROCM) && ROCM_VERSION < 40100)
++>>>>>>> new_upstream_branch

**Comment by zstreet87** [comment]
@jeffdaily
So you are referring to this change in Macros.h and want to take the upstream change?

++<<<<<<< HEAD
 +#if defined(__ANDROID__) || defined(__APPLE__) || defined(__HIP_PLATFORM_HCC__)
++=======
+ #if defined(__ANDROID__) || defined(__APPLE__) || defined(__XROS__) || \
+     (defined(USE_ROCM) && ROCM_VERSION < 40100)
++>>>>>>> new_upstream_branch

File: torch/csrc/CudaIPCTypes.cpp [hunk 1]
```diff
@@ -162,23 +162,23 @@ CudaIPCSentData::CudaIPCSentData(
     // because the main thread may have queued extra work on the stream, which
     // this event will consequently wait for (uselessly).
     cuda_ipc_global_entities.sync_events_used_ ++;
-    C10_CUDA_CHECK(cudaEventCreateWithFlags(
+    C10_HIP_CHECK(hipEventCreateWithFlags(
```
Comment (jithunnair-amd) [inline torch/csrc/CudaIPCTypes.cpp:165]: @zstreet87 Are these hipified changes being committed to the branch? If so, why? cc @jeffdaily

Comment (zstreet87) [reply torch/csrc/CudaIPCTypes.cpp:165]: goodness bad habit!  Only the few files I changed for resolving merge conflicts should have been pushed...

**Comment by zstreet87** [comment]
New error that I haven't seen before.  Therefore in the interest of time, this branch will be nuked and I will create another one with the same name and start over.

**Pull request closed without merging**